ara/exec/parser/manifest_parser.h

```diff
@@ -11,27 +12,74 @@ struct ExecutionManifest
 {
     // TODO add proper members
 
-    bool operator==(const ExecutionManifest&) const
+    bool operator==(const ExecutionManifest&) const noexcept
     {
         return true;
     }
+    bool operator!=(const ExecutionManifest& other) const noexcept
+    {
+        return !(*this == other);
+    }
 };
 
 struct MachineManifest
 {
     // TODO add proper members
 
-    bool operator==(const MachineManifest&) const
+    bool operator==(const MachineManifest&) const noexcept
     {
         return true;
     }
+    bool operator!=(const MachineManifest& other) const noexcept
+    {
+        return !(*this == other);
+    }
+};
+
+class Manifest  // Simple union wrapper- no std::variant in c++14
+{
+public:
+    enum class Type { kExecution, kMachine, kUnknown };
+
+public:
+    Manifest(){};
+    Type get_type() const
+    {
+        return type_;
+    }
+    ExecutionManifest get_execution() const
+    {
+        return exec_man;
+    }
+    MachineManifest get_machine() const
+    {
+        return mach_man;
+    }
+
+    void set_execution(const ExecutionManifest& man)
+    {
+        type_ = Type::kExecution;
+        exec_man = man;
+    }
+    void set_machine(const MachineManifest& man)
+    {
+        type_ = Type::kMachine;
+        mach_man = man;
+    }
+
+private:
+    Type type_{Type::kUnknown};
+    union
+    {
+        ExecutionManifest exec_man;
+        MachineManifest mach_man;
+    };
```

Comment:
I found out that I can't use union due to the fact that ExecutionManifest and MachineManifests are not trivially copy-able and this forbids me from using those structures as a union's elements. I changed this into following:
```cpp
private:
    Type type_{Type::Unknown};
    std::unique_ptr<ExecutionManifest> exec_man{};
    std::unique_ptr<MachineManifest> mach_man{};
```
And I obviously changed setters and getters in order to reflect those changes.
This increases size of `Manifest` class a bit (by one `std::unique_ptr`) and forces us to create those objects on heap but IMO this solution is just good enough for our purposes (Manifests are not parsed so often- only on Machine State's transitions)
I introduced changes mentioned above in the next part of implementation (which is not pushed yet), so in this PR we can leave it as it is.

Reply:
Z tego co sie zagrzebałem w ten dokument (sa w nim tez rozpisane interfejsy): 
https://www.autosar.org/fileadmin/user_upload/standards/adaptive/17-03/AUTOSAR_TPS_ManifestSpecification.pdf
to Application Manifest i Machine Manifest maja jedynie tyle wspolnego ze w pierwszym polu jest ich typ, reszta to totalnie dwa oddzielne byty. Zostawilbym Type w klasie bazowej, a cala reszte juz w klasach pochodnych rozpatrywal.

Reply:
Sorry ze mnie ostatnio nie było, ale mam wiecej pracy w projekcie. Sprobuje dzis wieczorem usiasc troche do tego projektu.

Reply:
No tak, są różne dlatego nie mają żadnej wspólnej klasy bazowej- to nie miałoby po prostu sensu, bo rzeczywiście nie mają żadnych pól wspólnych.
Aczkolwiek jak mamy plik .json to nie wiemy jaki typ manifestu to jest, więc musimy mieć jedną funkcję, która przyjmuje wszystkie jsony i w zależnosci od odczytanego typu, odpowienio je parsuje (jako ExecutionManifest albo MachineManifest). I tak to właśnie jest w tej chwili zrobione. Później w zależności od zwróconego przez `Manifest` typu (`Manifest::get_type()`) można wykonać dalsze kroki dla danego typu

Reply:
Hmm a co z wykorzystaniem wzorca Factory? Mamy klase Manifest ktora ma Type i scieżke do pliku .json. W metodzie parse odczytujemy Type i tworzymy klase pochodna odpowiedniego typu i tam juz rozczytujemy wszystkie pola?

Reply:
I co zwrócisz z tej fabryki? Wskaźnik do klasy bazowej, np. `Manifest`, po ktorej dziedziczy ExecutionManifest i MachineManifest. Nie dostaniesz się w ten sposób do pól odpowiednich dla konkretnego typu manifestu. 
Fabryke stosujemy kiedy mamy klasy pochodne, które mają publiczne interfejsy 1:1 takie jak klasa bazowa. Inaczej to nie ma sensu

Reply:
Przynajmniej tak rozumiem Twój pomysł. Jeżeli coś źle zinterpretowałem to mów śmiało- może razem uda nam się wypracować jakieś optymalne rozwiązanie :+1:

Reply:
Wyszedłem z założenia że wiemy tak naprawdę jaki manifest parsujemy, w końcu są one dostarczane przez producenta do każdego pojazdu i każdej aplikacji, a klasa Manifest ma nam uwspólnić wywołanie metody parse. Równie dobrze klasy manifestów mogłyby być osobne, bo nic wspólnego poza byciem .json'em nie mają.